UniExeterRSE/NetClamp
# [Ready for Review] Add basic polygon ROI drawing

Closes #145.

~Test by setting `draw_mode = 'POLY'`~:

Following steps should cover the basic functionality:
 - [X] Draw polygons
 - ~Add button to change mode between rectangle and polygon ROI drawing~ (or remove rect-rois completely)
 - [x] Submit a polygon ROI to the app backend
 - [x] Handle the polygon ROI creation in the backend
 - [x] Pass polygon points from backend to UI
 - [x] Draw polygons in the UI obtained from the backend
 - ~Add the polygon ROI to the chart (perhaps same current columns, but add type e.g. `poly(6)` for a hexagon)~ moved to #150.
 - [x] Update save/load ROIs functionality
 - [X] Gracefully cancel the polygon if:
   - [X] Try to create "polygon" with less than 3 sides
   - ~Change mode back to rect (if rect still exists)~
   - ~Other camera or ROI button pressed~
   - ~User wants to (how to implement? keypress? dedicated UI button?)~
   - [X] User clicks outside the canvas

Potential stretch goals:
 - Different colour for drawing vs fixed ROI (maybe red for drawing, then black when finished/closed)
 - Text indication of "polygon drawing in progress" beyond changing to crosshair
 - Add minimum polygon area to be accepted (user can still get a completely flat polygon, e.g. triangle `[[0, 0], [0, 20], [0, 40]]`)
 - Use a standardised format for the shape object, e.g. [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry)
